### DISCUSSION TITLE
Tab background on active/hovered is dark even in light mode

### DISCUSSION THREAD

**Discussion by Octal450** [microsoft/terminal#14844] [General]
### Windows Terminal version

1.16.10262.0

### Windows build number

10.0.22621.1194

### Steps to reproduce

* Open Windows terminal when the light them OR use system theme (with Windows set to light mode)
* Observe the tab row
![image](https://user-images.githubusercontent.com/25598604/217870576-73c8e4af-ab17-4237-aee1-683c967d286c.png)

* Hover over a tab while another tab is open:
![image](https://user-images.githubusercontent.com/25598604/217871527-454630ba-4b23-4a36-8aeb-9b5f9c15c7fe.png)

### Expected Behavior

Like in the previous version, the tabs should be light when in light mode (like Settings is)
![image](https://user-images.githubusercontent.com/25598604/217871047-a8970c9b-e81f-41aa-b2bb-6f52a07d71c8.png)

### Actual Behavior

The tabs (except for settings) are dark when in light mode.

**Comment by Octal450** [comment]
PS: I noticed that by default the application was now in "Dark" mode as opposed to "Windows default". Why? Shouldn't it track Windows' color scheme by default?

**Comment by CirnoT** [comment]
I believe this might be desired behavior. Tab color now tracks background color of selected color scheme for given profile.

![WindowsTerminal_2023-02-10_01-31-36](https://user-images.githubusercontent.com/1447794/217970238-b0ffd914-c411-4e69-8898-596af57389e0.png)
![WindowsTerminal_2023-02-10_01-32-13](https://user-images.githubusercontent.com/1447794/217970242-a56aea11-9822-4070-9dce-1fa73d2ba942.png)
![WindowsTerminal_2023-02-10_01-32-27](https://user-images.githubusercontent.com/1447794/217970243-e1f2e334-7df2-4789-b55c-072b62a50387.png)

Can't say I particularly agree with this choice but I also very much doubt it's gonna be reverted or changed.

**Comment by j4james** [comment]
This behavior is configurable with the new theme settings. See the documentation here:
https://learn.microsoft.com/en-us/windows/terminal/customize-settings/themes

**Comment by Octal450** [comment]
That's rather silly... I don't understand the reasoning for such functionality at all. Especially being on by default...

**Comment by 3tothe6** [comment]
This may be a bug introduced in v1.16 in which theme support was stabilized. For people who don't have time reading the doc, backup your settings.json and revert to [v1.15.3465.0](https://github.com/microsoft/terminal/releases/tag/v1.15.3465.0), then your problem may be solved.

**Comment by CirnoT** [comment]
> This behavior is configurable with the new theme settings. See the documentation here: https://learn.microsoft.com/en-us/windows/terminal/customize-settings/themes

~It is actually impossible to redefine built-in themes as they are removed from JSON file; so "Use system theme" behavior can not be reproduced.~

For those looking to reproduce old behavior; see attached theme configuration:

```json
{
    "themes": 
    [
        {
            "name": "Dark (legacy)",
            "tab": 
            {
                "background": null,
                "unfocusedBackground": null
            },
            "window": 
            {
                "applicationTheme": "dark"
            }
        },
        {
            "name": "Light (legacy)",
            "tab": 
            {
                "background": null,
                "unfocusedBackground": null
            },
            "window": 
            {
                "applicationTheme": "light"
            }
        },
        {
            "name": "Use Windows theme (legacy)",
            "tab": 
            {
                "background": null,
                "unfocusedBackground": null
            }
        }
    ]
}
```

**Comment by zadjii-msft** [comment]
As noted in this thread, we introduced support for [Themes](https://learn.microsoft.com/en-us/windows/terminal/customize-settings/themes) in 1.16 (and further refined them in 1.17). As a part of that work, we took the opportunity to revisit what the "out-of-the-box" theme of the Terminal looks like[^1]. We really wanted to revisit the fact that for most users (who haven't set the OS theme), when they launch the Terminal, they get a window that's mostly _black,_ with a white titlebar. This contrast seemed a little silly to us. So we opted to optimize for the first run experience being dark theme, [a dark-gray titlebar with tabs that were seamless (effectively, black)](https://i.redd.it/k1p5j6p7r3kz.jpg). 

Of course, customizability has been one of my zealous ideals here on the Terminal, so while changing the default themes, I wanted to make sure to give folks whatever knobs they needed to make the Terminal look however they like.

The situation we probably failed to consider as deeply as we should was folks who had a light titlebar, and wanted to just use the new "light" theme out of the box. As noted in this thread - the default light theme with the default dark colorscheme creates a black terminal, with a black tab, on a white titlebar. Clearly, there's a lot of folks here who don't love this look[^2]. 

I've maybe overestimated in my head how willing folks would be to change the theme settings if they didn't like the defaults. That's on me - I'm always tinkerering with my settings, but my experience is not universal. Most folks just want sane defaults. I had made a lot of assumptions and let my personal preferences get involved here, so that's my bad. 

I think we should more broadly promote the "legacy" themes as authored by @CirnoT in this thread. Those are just about what the old themes looked like, so if that's what people want, I wanna give the people what they want. I'd like to leave this thread open to track ingesting those into the default list of themes. Now, that might be a bit tricky. The built-in theme names need to get localized, and I don't know if we can check in loc changes like this in a servicing release. If we can't, then they'd be first (re)-available in 1.17 Stable/1.18 Preview. 

We're discussing some options as a team, but probably won't have more to share till after we discuss more on ~Monday~[^3]. 

In the meantime, I'd recommend everyone who's missing the old themes insert that blob into their [settings.json](https://github.com/microsoft/terminal/wiki/Frequently-Asked-Questions-(FAQ)#where-can-i-find-the-settings-file).

Thanks for all the lively feedback here!

[^1]: This had already changed slightly over the last couple releases, as we were working on ingesting some of the WinUI changes for Windows 11.

[^2]: Though, I do think this is where we're getting into a discussion of UX _taste_ - in discussion we had as a team, we generally preferred the new light theme with the seamless black tab, over the old "black terminal, white tab, light titlebar" look of the old theme. Though the white tab/light titlebar does look a lot like the vintage console...

[^3]: EDIT: some conflicts came up and our team sync is getting moved this week, though I'm not quite sure when it'll be yet.

**Reply by PinkSerenity** [reply]
> We really wanted to revisit the fact that for most users (who haven't set the OS theme), when they launch the Terminal, they get a window that's mostly _black,_ with a white titlebar. This contrast seemed a little silly to us. So we opted to optimize for the first run experience being dark theme, [a dark-gray titlebar with tabs that were seamless (effectively, black)](https://i.redd.it/k1p5j6p7r3kz.jpg).
> 
> Of course, customizability has been one of my zealous ideals here on the Terminal, so while changing the default themes, I wanted to make sure to give folks whatever knobs they needed to make the Terminal look however they like.

I completely understand your reasoning here, however, I want to raise concern regarding one of the core reasons for why theming exists which is largely forgotten in such discussions: accessibility.

Just as there are people who are unable to use a light theme, some people might not recognize text and iconography when using a dark theme. That more and more apps support a "Use System Theme" setting is a boon for such people. It's just a pain if this isn't the default because of aesthetic concerns by the developers or designers.

For the record, I think Microsoft makes it unreasonably hard for impaired users to setup their PC, since choosing a theme is impossible until they have access to the settings. Android is similar, the only difference is that since Android 13 dark mode is enforced.

But this shall only focus on the app, the operating system is an entirely different matter. I for one would welcome a change back to making "Use System Theme" as default. I don't think that it's an issue if the color scheme itself is still dark, because if the UI theme is light, impaired users can easily change it.

Thank you very much for your consideration!

**Comment by Octal450** [comment]
@zadjii-msft ,
Thank you for the detailed reply and time taken. I appreciate it greatly.

I do understand your reasoning, especially given that the dark mode has taken off in popularity (especially among techies like us) quite greatly over the last few years.

However, the issue is with a system in light mode, having one window with a dark titlebar looks horribly out of place, while all the other windows regardless of color have a white titlebar - which fits the white UI of most applications on a system with light mode selected (which I would like to point out, is the default that Windows ships with). 

Not to mention the consistency nightmare of a light titlebar with dark tabs when "light" mode is selected.

Personally, I don't see the issue of a light titlebar with a dark command window, given that it has been this way for roughly 25 years. I strongly believe that respecting the operating system's defaults should be the highest priority in order to provide a seamless and consistent user experience.

Kind Regards,
Josh

PS: Apologies for the close - misclicked.

**Comment by zadjii-msft** [comment]
> I strongly believe that respecting the operating system's defaults should be the highest priority in order to provide a seamless and consistent user experience.

You seem like the kind of person who might like what we've got in store in 1.17: _(warning, this gif was taken on a debug build - it's faster on a release build)_

![](https://user-images.githubusercontent.com/18356694/214559480-7afa0de8-5123-4776-bd8d-e9027eb03ab5.gif)

**Comment by CirnoT** [comment]
> I've maybe overestimated in my head how willing folks would be to change the theme settings if they didn't like the defaults. That's on me - I'm always tinkerering with my settings, but my experience is not universal. Most folks just want sane defaults. I had made a lot of assumptions and let my personal preferences get involved here, so that's my bad.

Speaking from experience - I was very much surprised to see Terminal suddenly changed to a dark theme and sought out to see what caused it. While investigating and switching to light theme, the initial impression I had when tab color remained dark is that there is something wrong.

Note that updates for Terminal are now completely seamless and happen in background. I have not even rebooted my PC and was very much unaware of the update happening until I checked it. Maybe this is part of the issue here, especially with switching existing users to dark with no update prompt or heads-up (a simple "What's new" message on first run might've been sufficient to clear some misunderstandings)

**Comment by Octal450** [comment]
Hi,
That does look nifty.

Personally, I am not so worried about the color of the terminal body itself, especially being used to a dark terminal from all my years of using various operating systems (except macOS... which has a light one)

For me personally, it was more regarding the titlebar/tabs/window decorations. Working in the light mode environment develops "reflex" like responses when wanting to resize, close, minimize, etc as a result of the window decorations being consistent regardless of the application (well... except special programs which provide their own, but you get the idea). 

As a result, I find myself fumbling now and then using the terminal in dark mode, being the sole window who's titlebar has a different appearance without using a custom title bar. 

An interesting sidenote, it also seems the new Photos app in Windows 11 has the exact same - a default of dark mode rather than respecting the color settings.

Kind Regards,
Josh

**Comment by fxzxmic** [comment]
@zadjii-msft I know that you want to provide a seamless experience of tabs and interfaces, and you have done it. This is a great idea.
However, I hope to consider the diversity of user needs. And the dark tabs and light tabs' bar actually aggravate the sense of fragmentation. Fortunately, it does not cause much trouble because it retains the old experience.

Thank @CirnoT for the method to restore the old experience.

**Comment by Limyx826** [comment]
Suprise me when the first thing that happen when I open Terminal is seeing dark theme. Even changing into light theme also have some of the dark theme elements left over.

**Comment by jcefoli** [comment]
I use Dark Windows mode but Light app mode, and this looks really bad. I know it may be customizable now, but you really shouldn't be changing people's settings because a feature is improved. I get the tabs may be customizable now, but I don't know how to do it and really think it should look like how it did before this upgrade for those of us using light app mode.

![image](https://user-images.githubusercontent.com/6618013/219196934-837aaa0c-b5f3-497c-9923-7a4d55187ef5.png)

**Comment by OS-Scratch** [comment]
When using PowerShell, the tabs work as expected. However, in cmd, even though I use light mode, the tab stays dark. I believe it’s caused by the theme of the current tab that’s open.
